### DISCUSSION TITLE
Is it possible to run r-languageserver with JupyterHub in separate environments?

### DISCUSSION THREAD

**Discussion by jrouly** [REditorSupport/languageserver#605] [Q&A] [answered]
I'm not 100% sure if this is the correct repository to file this issue so please forgive me in advance if I've bungled this.

I'm trying to run JupyterHub with an R kernel, using Conda (micromamba) to manage environments, with an R language server.

The steps I've followed are essentially:
1. Install `jupyterlab`, `jupyterhub` into the `jupyter` environment
2. Install `R`, `r-irkernel`, `r-languageserver` into the `R` environment
3. Activate the `R` environment and invoke `IRkernel::installspec(user = FALSE, prefix = '$JUPYTER_HOME')`

This works great and sets up an R kernel. However, despite having installed `r-languageserver` (which I can confirm by running `library(languageserver)` in a notebook), JupyterLab is unable to detect the R language server. It complains:

```
No language server for r detected
```

and recommends I troubleshoot finding `Rscript` on the `PATH`. It _is_ on the PATH (in the `R` environment), and JupyterLab is clearly aware of this since it's able to run notebooks on the R kernel just fine. And yet, it cannot find the language server.

**Comment by randy3k** [comment, marked as answer]
I think [jupyter-lsp](https://github.com/jupyter-lsp/jupyterlab-lsp) may be a better place to ask just a question because it is related to the environmental setup of jupyterhub.

**Reply by jrouly** [reply]
Thanks, @randy3k. I went ahead and filed [an issue](https://github.com/jupyter-lsp/jupyterlab-lsp/issues/923) over there. I'll call this resolved over here :)
